pyogrio/tests/test_geopandas_io.py

```diff
@@ -342,12 +371,31 @@ def test_read_write_datetime_tz_with_nulls(tmp_path, use_arrow):
         {"dates": dates, "geometry": [Point(1, 1), Point(1, 1), Point(1, 1)]},
         crs="EPSG:4326",
     )
-    fpath = tmp_path / "test.gpkg"
+    fpath = tmp_path / f"test{ext}"
     write_dataframe(df, fpath, use_arrow=use_arrow)
     result = read_dataframe(fpath, use_arrow=use_arrow)
     if use_arrow:
-        # with Arrow, the datetimes are always read as UTC
-        df["dates"] = df["dates"].dt.tz_convert("UTC")
+        if ext == ".fgb":
+            # when FlatGeoBuffer is read with Arrow, for datetimes with equal timezone,
+            # a column type with the appropriate minutes offset is returned.
+            # REMARK: For .fgb, the timezone is just dropped when reading or writing!!!
+            #     -> 2020-01-01T09:00:00.123-05:00 becomes 2020-01-01T09:00:00.123
+            df["dates"] = df["dates"].dt.tz_localize(tz=None)
+        elif ext in (".geojson", ".geojsonl"):
+            # when GeoJSON is read with Arrow, for datetimes with equal timezone, a
+            # column type with the appropriate minutes offset is returned.
+            # REMARK: for .geojson, the data is written fine, but when reading it goes
+            # wrong: 2020-01-01T09:00:00.123-05:00 becomes 2020-01-01T04:00:00.123-05:00
```

Comment:
I think this was a bug that is fixed in the meantime? (so will have to update the expected result for that depending on the GDAL version)